### PR TITLE
Currency channel context

### PR DESCRIPTION
**Pull request by marekrzytki**
Add currency or first channel context to always retrieve channel and avoid stage problems

### CODE REVIEW

File: tests/Application/src/Entity/Channel.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\Channel as BaseChannel;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+
+final class Channel extends BaseChannel implements ChannelInterface
+{
+    public function getAllCurrencies(): Collection
+    {
+        $allCurrencies = new ArrayCollection([
```
Comment (senghe) [inline tests/Application/src/Entity/Channel.php:24]: Is there any chance to have base currency same as one from the `getCurrencies()` method? If so, it seems we need the de-duplifying condition

File: tests/Application/src/Entity/Channel.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+use BitBag\SyliusMolliePlugin\Entity\ChannelInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\Channel as BaseChannel;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+
+final class Channel extends BaseChannel implements ChannelInterface
+{
+    public function getAllCurrencies(): Collection
```
Comment (senghe) [inline tests/Application/src/Entity/Channel.php:22]: And why we need this in test application? Can't we put this into trait and use the trait here?

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Application/src/Entity/Channel.php (hunk 1 above)
Comment (marekrzytki) [reply tests/Application/src/Entity/Channel.php:24]: There is no chance. Base currency is other object than are stored in getCurrencies collection

File: tests/Application/src/Entity/Channel.php (hunk 2 above)
Comment (marekrzytki) [reply tests/Application/src/Entity/Channel.php:22]: Yes sure, I will move it to trait

**Review by senghe — APPROVED** [review]

**Pull request merged**